src/lib.rs

```diff
@@ -2439,6 +2456,38 @@ impl Build {
         Ok(())
     }
 
+    fn add_inherited_rustflags(&self, cmd: &mut Tool, target: &Target) -> Result<(), Error> {
+        let env_os = match self.getenv("CARGO_ENCODED_RUSTFLAGS") {
+            Some(env) => env,
+            // No encoded RUSTFLAGS -> nothing to do
+            None => return Ok(()),
+        };
+
+        let mut cc_flags: Vec<OsString> = env_os
+            .to_string_lossy()
+            .split("\u{1f}")
+            // Strip prefixes from rustc flags
+            .flat_map(|flag| {
+                if flag == "-Z" || flag == "-C" {
+                    None
+                } else if flag.starts_with("-Z") || flag.starts_with("-C") {
+                    Some(&flag[2..])
+                } else {
+                    Some(flag)
+                }
+            })
+            .flat_map(|flag| rustc_to_cc_flag(flag, cmd.family))
```

Comment:
I see the appeal in the simplicity of this, but I think this approach of ignoring where a flag came from is wrong, we should be much more strict.

For example, `rustc --cfg branch-protection="yes"` would translate to the flag `-mbranch-protection="yes"`, which is completely bogus.

Reply:
I did also consider an approach where we merge the flag prefixes with flags and then match on flags like `-Clto` and `-Zbranch-protection`. The downside there is that e.g. with branch protection, once the flag gets stabilised it'll stop matching. I guess we could get around it by matching it on either -Z or -C. Should I switch back to the more strict approach then?